### PR TITLE
WIP: Cleanup `SQLiteStorage`

### PR DESCRIPTION
**Pull request by palango**
Put all data modification statement in transactions and remove unused
methods.

### CODE REVIEW

File: raiden/storage/sqlite.py [hunk 1]
```diff
@@ -778,52 +781,6 @@ def get_state_changes(self, limit: int = None, offset: int = None) -> List[str]:
         entries = self._get_state_changes(limit, offset)
         return [entry.data for entry in entries]
 
-    def get_snapshots(self) -> List[SnapshotEncodedRecord]:
```
Comment (konradkonrad) [inline raiden/storage/sqlite.py:781]: Why was this removed?

Comment (palango) [reply raiden/storage/sqlite.py:781]: Because it's not used anywhere
